index.js

```diff
@@ -2,19 +2,23 @@ const isString = x => typeof x === 'string'
 const isArray = x => Array.isArray(x)
 const isObject = x => typeof x === 'object' && !isArray(x)
 
-const clean = (a, b) => (!b ? a : isString(b[0]) ? a.concat([b]) : a.concat(b))
-const child = x => f =>
-  x || x === 0 ? (isArray(x) ? x.reduce(clean, []).map(f) : x + '') : []
+const clean = (arr, n) =>
+  (n && Array.prototype.push.apply(arr, (isString(n[0]) ? [n] : n)), arr)
```

Comment:
`concat` will create a new array on each call, `push` does not.

Reply:
FYI 3 viable options here:

- `Array.prototype.push.apply` correct, low mem, but most bytes
- `[].prototype.push.apply` correct, slightly more mem?? least bytes
- `arr.prototype.push.apply` correct but a bit confusing, low mem, few bytes

granted - maybe it doesn't matter, either!!

Reply:
What you have makes most sense to me, good to see the options though. I am not too concerned about bytes here. So all good :+1: